fix: opt-in server.fs.cachedChecks (#17807)
vitejs/vite: docs/config/server-options.md

Blocklist for sensitive files being restricted to be served by Vite dev server. This will have higher priority than [`server.fs.allow`](#server-fs-allow). [picomatch patterns](https://github.com/micromatch/picomatch#globbing-features) are supported.

## server.fs.cachedChecks

- **Type:** `boolean`
- **Default:** `false`
- **Experimental**

> The `fs.cachedChecks` optimization caches filenames of accessed directories to avoid repeated filesystem operations. In Windows in particular, this could result in a performance boost. It is disabled by default because there are edge cases when writing a file in a cached folder and immediately importing it.

Reviewer: ```suggestion
Caches filenames of accessed directories to avoid repeated filesystem operations. Particularly in Windows, this could result in a performance boost. It is disabled by default due to edge cases when writing a file in a cached folder and immediately importing it.
```

Shortening it a little 😅
